extensions/sql-database-projects/src/models/options/deployOptionsModel.ts

```diff
@@ -8,250 +8,64 @@ import * as mssql from 'mssql';
-			, this.deploymentOptions.ignoreTablePartitionOptions.displayName
-			, this.deploymentOptions.doNotEvaluateSqlCmdVariables.displayName
-			, this.deploymentOptions.disableParallelismForEnablingIndexes.displayName
-			, this.deploymentOptions.disableIndexesForDataPhase.displayName
-			, this.deploymentOptions.restoreSequenceCurrentValue.displayName
-			, this.deploymentOptions.rebuildIndexesOfflineForDataPhase.displayName
-			, this.deploymentOptions.preserveIdentityLastValues.displayName
-			, this.deploymentOptions.isAlwaysEncryptedParameterizationEnabled.displayName
-			, this.deploymentOptions.allowExternalLibraryPaths.displayName
-			, this.deploymentOptions.allowExternalLanguagePaths.displayName
-			, this.deploymentOptions.hashObjectNamesInLogs.displayName
-			, this.deploymentOptions.doNotDropWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreWorkloadClassifiers.displayName
-			, this.deploymentOptions.ignoreDatabaseWorkloadGroups.displayName
-			, this.deploymentOptions.doNotDropDatabaseWorkloadGroups.displayName].sort();
+	public UpdateOptionsMapTable() {
+		this.optionsMapTable = this.deploymentOptions.optionsMapTable;
 	}
 
 	/**
-	 * Gets the options checkbox check value
-	 * @returns string[][]
+	 * Gets the options checkbox values by iterating through the labels and gets the default/changed value from the optionsMapTable
+	 * Returns data as [optionName, booleanValue], where these values are sending to the options table component
 	 */
 	public getOptionsData(): string[][] {
 		let data: any = [];
-		this.optionsLookup = {};
+		this.optionsLookup = new Map<string, boolean>();
 		this.optionsLabels.forEach(l => {
-			let checked: boolean = this.getDeployOptionUtil(l);
-			data.push([checked, l]);
-			this.optionsLookup[l] = checked;
+			let checked: boolean | undefined = this.getDeployOptionUtil(l);
+			if (checked !== undefined) {
+				data.push([checked, l]);
+				this.optionsLookup?.set(l, checked);
+			}
 		});
 		return data;
 	}
 
-	public setDeploymentOptions() {
-		for (let option in this.optionsLookup) {
-			this.setDeployOptionUtil(option, this.optionsLookup[option]);
-		}
-	}
-
 	/*
-	* Sets the selected/changed value of the option
+	* Sets the selected option checkbox value to the optionsMapTable
 	*/
-	public setDeployOptionUtil(label: string, value: boolean) {
-		this.optionsMapTable[label].value = value;
+	public setDeploymentOptions(): void {
+		for (let option of this.optionsLookup) {
+			let val = this.optionsMapTable[option[0]];
+			if (val !== undefined && val?.value !== option[1]) {
+				val.value = option[1];
+				this.optionsMapTable[option[0]] = val;
+			}
+		}
+
+		// Set the deployment optionsMapTable with the updated optionsMapTable
+		this.deploymentOptions.optionsMapTable = this.optionsMapTable;
 	}
 
 	/*
 	* Gets the selected/default value of the option
 	*/
-	public getDeployOptionUtil(label: string): boolean {
+	public getDeployOptionUtil(label: string): boolean | undefined {
```

Comment:
This function name isn't very clear - what is a "Deploy Option Util"? Functions that start with `get...` should by convention be returning a single thing that they're retrieving.